smac_planner/src/smac_planner.cpp

```diff
@@ -239,6 +262,11 @@ nav_msgs::msg::Path SmacPlanner::createPlan(
   steady_clock::time_point a = steady_clock::now();
 #endif
 
+  // Initialize (if needed) costmap used for planning
+  if (!_costmap) {
+    initCostmap();
```

Comment:
This isn't initialization, it will update over time and should happen at every cycle

Reply:
So I think you should just be calling this outright and then the costmap that's downsampled doesn't need to be stored as a class variable, it can be local to this function and used by the rest of the functions and deleted at the end of the planning call.

Reply:
Yes, at some point I had it like that, but then (wrongfully) thought it could actually just be called once at the beginning. I'll move it back to doing the call everytime.